src/GraphQL/ObjectExtensions.cs

```diff
@@ -103,6 +103,17 @@ public static object ToObject(this IDictionary<string, object> source, Type type
         /// <remarks>There is special handling for strings, IEnumerable&lt;T&gt;, Nullable&lt;T&gt;, and Enum.</remarks>
         public static object GetPropertyValue(this object propertyValue, Type fieldType, IGraphType mappedType = null)
         {
+            bool SkipEnumerable()
+            {
+                if (fieldType == typeof(string))
+                    return true;
+
+                if (fieldType == typeof(byte[]) && propertyValue is string)
```

Comment:
I think this can be safely considered a special case.

Reply:
Good idea.

Reply:
If we go further, we can make the assertion that `IEnumerable` has nothing to do with it, we can simply move `ValueConverter.ConvertTo(value, fieldType);`/lookup to the beginning of the method.

Reply:
Done.